alacritty/src/input.rs

```diff
@@ -710,6 +710,10 @@ impl<'a, T: EventListener, A: ActionContext<T> + 'a> Processor<'a, T, A> {
 
                 // Don't suppress when there has been a `ReceiveChar` action
                 *suppress_chars.get_or_insert(true) &= binding.action != Action::ReceiveChar;
+
+                if let Action::Esc(_) = binding.action {
+                    self.ctx.clear_selection();
+                }
```

Comment:
Can't we move this into the `execute` function?

Reply:
We don't want all the bindings to reset selection. For example, SrollPageUp/ScrollPageDown shouldn't reset selection imho (or I wouldn't be able to select text on a laptop), so thats an idea of making it for `Esc` only.

Reply:
> Oh, are you about just moving clear_selection to execute's Action::Esc arm?

Yes.